src/library.cpp

```diff
@@ -3,10 +3,27 @@
 
 #include <kiwix/manager.h>
 #include <kiwix/tools.h>
+#include <zim/error.h>
 
 #include <QtDebug>
 #include <QtConcurrent/QtConcurrentRun>
 
+zim::Entry getArchiveEntryFromUrl(const zim::Archive& archive, const QUrl& url)
+{
+    std::string path = url.path().toUtf8().constData();
+    if (path[0] == '/')
+        path = path.substr(1);
+
+    try {
+        return archive.getEntryByPath(path);
+    } catch (zim::EntryNotFound& e) {
+        if (path.empty() || path == "/") {
+            return archive.getMainEntry();
+        }
+    }
+    throw zim::EntryNotFound("Cannot find entry for non empty path");
+}
+
```

Comment:
I think you can preserve this function in `src/urlschemehandler.cpp` as an enhancement of `getEntryFromPath()` (this way the change will be more conspicuous).

Reply:
Done